packages/manager/src/queries/cloudpulse/alerts.ts

```diff
@@ -245,9 +245,35 @@ export const useServiceAlertsMutation = (
       return updateServiceAlerts(serviceType, entityId, payload);
     },
     onSuccess() {
+      const allAlerts = queryClient.getQueryData<Alert[]>(
+        queryFactory.alerts._ctx.all().queryKey
+      );
+
+      const entityAlertIds =
+        allAlerts
+          ?.filter((alert) => alert.entity_ids.includes(entityId))
+          .map((alert) => alert.id) || [];
+
```

Comment:
```suggestion
    onSuccess(_, payload) {
```
just utilize payload, since it has all the alert ids just iterate it and invalidate its query

Reply:
payload doesn't have all alert ids, it only has enabled alert ids, so i am just invalidating all the alerts specific to an entity because this api call updates that entity_id's res obj. One other option is to calculate the difference between old and new alerts but instead of doing that computation, we can just invalidate entity id specific alerts.